### PR TITLE
Adds a max clip value to replace the max allowed min max ratio

### PR DESCRIPTION
**Pull request by unconst**
This PR replaces the max allowed min-max ratio computation in exchange for a max clip check. 
The max clip value checks that the normalized weights do not exceed the max clip value.

A new hyper-parameter MaxClipValue has been added with an initial value equal to u32:MAX = 4,294,967,295
At this initial value, there is no restriction on the u32 weights set by a validator.

Merge With Vunes PR:

This PR adds 3 new hyperparameters as sudo-protected constants to the subtensor pallet.

ScalingLawPower: u8 = 50
SynergyScalingLawPower: u8 = 60
ValidatorExcludeQuantile: u8 = 10
Each value is stored as an unsigned 8-bit integer
Each value x here represents the value x * 10^-2
Meaning each can store floats between [0.0, 1.0] with a precision of 0.01

### CODE REVIEW

**Review by shibshib — APPROVED** [review]

File: pallets/subtensor/src/weights.rs [hunk 1]
```diff
@@ -61,25 +61,51 @@ impl<T: Config> Pallet<T> {
         return false;
     }
 
-    pub fn check_length( uid: u32, uids: &Vec<u32>, weights: &Vec<u32>) -> bool {
+    // Returns true if the peer is setting a single self weight.
+    pub fn is_self_weight( uid: u32, uids: &Vec<u32>, weights: &Vec<u32> ) -> bool {
+        if weights.len() != 1 {
+            return false;
+        }
+        if uid != uids[0] {
```
Comment (opentaco) [inline pallets/subtensor/src/weights.rs:69]: Where is it ensured that the weight submitter UID is put first in the `uids` vector?

In Bittensor `convert_weights_and_uids_for_emit()` does not explicitly put the submitter UID first in the submitted weights+uids tensors.

File: pallets/subtensor/tests/weights.rs [hunk 2]
```diff
@@ -120,6 +120,34 @@ fn test_weights_err_has_duplicate_ids() {
...
+		run_to_block( 2 );
+    	let _neuron = register_ok_neuron( 1, 1);
+		run_to_block( 3 );
+		let _neuron = register_ok_neuron( 2, 2);
+		run_to_block( 4 );
+    	let _neuron = register_ok_neuron( 3, 3);
+		run_to_block( 5 );
+    	let _neuron = register_ok_neuron( 4, 4);
+
+		Subtensor::set_max_weight_limit(u32::MAX/5); // Set max to u32::MAX/2
```
Comment (opentaco) [inline pallets/subtensor/tests/weights.rs]: There is an inconsistency here between `u32::MAX/5` and `u32::MAX/2` in the comment.

File: pallets/subtensor/tests/weights.rs [hunk 3]
```diff
@@ -120,6 +120,34 @@ fn test_weights_err_has_duplicate_ids() {
...
+
+		// Non self weight fails.
+		let weights_keys: Vec<u32> = vec![1, 2, 3, 4]; 
+		let weight_values: Vec<u32> = vec![1, 1, 1, 1]; // normalizes to u32::MAX/2
```
Comment (opentaco) [inline pallets/subtensor/tests/weights.rs]: Just to clarify, what does `normalizes to u32::MAX/2` mean here and how is `u32::MAX/2` reached here?

**Review by Eugene-hu — APPROVED** [review]

File: pallets/subtensor/tests/weights.rs (hunk 3 above)
Comment (unconst) [reply pallets/subtensor/tests/weights.rs]: Both comments are incorrect.

**Review by opentaco — APPROVED** [review]

**Comment by shibshib** [comment]
Guys is this good to go? been open for a while.

File: validator_hyperparameters.md [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
+| **maxAllowedUids**                 | 4096                 |
+| **minAllowedWeights**              | 1024                 |
+| **rho**                            | 10                   |
+| **stakePruningDenominator**        | 20                   |
+| **stakePruningMin**                | 1024                 |
+| **targetRegistrationsPerInterval** | 2                    |
+| **validatorBatchSize**             | 32                   |
+| **validatorEpochLen**              | 250                  |
+| **validatorEpochsPerReset**        | 60                   |
+| **validatorSequenceLength**        | 128                  |
```
Comment (camfairchild) [inline validator_hyperparameters.md:20]: Add new hyperparameters here also

**Review by camfairchild — CHANGES_REQUESTED** [review]

**Review by camfairchild — APPROVED** [review]
